modules/thermal_hydraulics/src/physics/THMVACESinglePhaseFlowPhysics.C

```diff
@@ -0,0 +1,662 @@
+      params.set<AuxVariableName>("variable") = PRESSURE;
+      params.set<std::vector<SubdomainName>>("block") = flow_channel->getSubdomainNames();
+      params.set<std::vector<VariableName>>("e") = {SPECIFIC_INTERNAL_ENERGY};
+      params.set<std::vector<VariableName>>("v") = {SPECIFIC_VOLUME};
+      params.set<UserObjectName>("fp") = _fp_name;
+      _sim->addAuxKernel(class_name, genName(comp_name, "pressure_uv_auxkernel"), params);
+    }
+    {
+      std::string class_name = "TemperatureAux";
+      InputParameters params = _factory.getValidParams(class_name);
+      params.set<AuxVariableName>("variable") = TEMPERATURE;
+      params.set<std::vector<SubdomainName>>("block") = flow_channel->getSubdomainNames();
+      params.set<std::vector<VariableName>>("e") = {SPECIFIC_INTERNAL_ENERGY};
+      params.set<std::vector<VariableName>>("v") = {SPECIFIC_VOLUME};
+      params.set<UserObjectName>("fp") = _fp_name;
+      _sim->addAuxKernel(class_name, genName(comp_name, "T_auxkernel"), params);
+    }
+
+    {
+      std::string class_name = "SpecificTotalEnthalpyAux";
+      InputParameters params = _factory.getValidParams(class_name);
+      params.set<AuxVariableName>("variable") = SPECIFIC_TOTAL_ENTHALPY;
+      params.set<std::vector<SubdomainName>>("block") = flow_channel->getSubdomainNames();
+      params.set<std::vector<VariableName>>("rhoA") = {RHOA};
+      params.set<std::vector<VariableName>>("rhoEA") = {RHOEA};
+      params.set<std::vector<VariableName>>("p") = {PRESSURE};
+      params.set<std::vector<VariableName>>("A") = {AREA};
+      _sim->addAuxKernel(class_name, genName(comp_name, "H_auxkernel"), params);
+    }
+  }
+}
+
+void
+THMVACESinglePhaseFlowPhysics::addUserObjects()
+{
+  const std::string class_name = "ADNumericalFlux3EqnHLLC";
+  InputParameters params = _factory.getValidParams(class_name);
+  params.set<UserObjectName>("fluid_properties") = _fp_name;
+  params.set<MooseEnum>("emit_on_nan") = "none";
+  params.set<ExecFlagEnum>("execute_on") = {EXEC_INITIAL, EXEC_LINEAR, EXEC_NONLINEAR};
+  _sim->addUserObject(class_name, _numerical_flux_name, params);
+}
+
+void
+THMVACESinglePhaseFlowPhysics::addInletBoundaries()
+{
+  for (const auto i : index_range(_inlet_components))
+  {
+    // Get component, which has reference to the controllable parameters
+    const auto & comp_name = _inlet_components[i];
+    const auto & comp = _sim->getComponentByName<PhysicsFlowBoundary>(comp_name);
+    UserObjectName boundary_numerical_flux_name = "invalid";
+    const auto & boundary_type = _inlet_types[i];
+
+    // Boundary fluxes should be updated as often as possible
+    ExecFlagEnum userobject_execute_on(MooseUtils::getDefaultExecFlagEnum());
+    userobject_execute_on = {EXEC_INITIAL, EXEC_LINEAR, EXEC_NONLINEAR};
+
+    // boundary flux user object
+    // we add them in addBCs for convenience
+    if (boundary_type == InletTypeEnum::MdotTemperature)
```

Comment:
Would someone be able to add a new BC formulation in their application without modifying THM?

Reply:
yes, by using a XYZFlowChannel instead of the Physics one
or by creating a new physics